hypertrace-gradle-java-convention-plugin/src/main/java/org/hypertrace/gradle/java/convention/JavaConventionPlugin.java

```diff
@@ -12,15 +12,23 @@
 public class JavaConventionPlugin implements Plugin<Project> {
 
   public void apply(Project target) {
+    // plugin meant to be applied at root level, it configures for all nested subprojects
     target
-        .getPluginManager()
-        .withPlugin(
-            "java",
-            unused -> {
-              createConvention(target);
-              configureCompatibility(target);
-              configureToolchain(target);
-              modifyTestJvmArgs(target);
+        .getSubprojects()
+        .forEach(
+            subproject -> {
+              subproject
```

Comment:
```suggestion
        .allprojects(
            project -> {
              project
```

A few things here:
1. we want to apply to the root, too - if it's java. A simple gradle project that isn't using multiple modules would be missed if we only applied to subprojects
2. the allprojects/subprojects(configure) form is I believe the idiomatic way (vs getSubprojects().forEach...
3. this does mean no opt-out. Hope that's ok. We can always find a way to do so if needed.